### PR TITLE
libspatialite: bump dependencies

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
v2 pipeline:

```
The v2 pipeline failed. Please, review the errors and note this will be required for pull requests to be merged in the near future.

See details:
Changes not allowed in build 1:

[libmysqlclient, libspatialite]
Only one library can be changed in the same PR.
```

?

**Comment by SpaceIm** [comment]
waiting for https://github.com/conan-io/conan-center-index/pull/19117

**Review by jwillikers — APPROVED** [review]

**Comment by RubenRBS** [comment]
Any update on this one @SpaceIm?

**Comment by SpaceIm** [comment]
It's related to https://github.com/conan-io/conan/issues/13603 I think.

**Review by jwillikers — APPROVED** [review]

**Comment by RubenRBS** [comment]
I've added a zlib range to the PR

**Review by czoido — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Comment by ericLemanissier** [comment]
The conflict should be resilved now, with the latest commit merged in master. Can we retrigger this one?

This pr blocks:
- #20616

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/6627845767)

File: recipes/libspatialite/all/conanfile.py [hunk 1]
```diff
@@ -85,28 +85,29 @@ def layout(self):
         basic_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("sqlite3/3.42.0")
-        self.requires("zlib/1.2.13")
+        # sqlite.h is included by spatialite/sqlite.h public header
+        self.requires("sqlite3/3.42.0", transitive_headers=True)
```
Comment (ericLemanissier) [inline recipes/libspatialite/all/conanfile.py:89]:
```suggestion
        self.requires("sqlite3/3.42.2", transitive_headers=True)
```
Fixes conflict with proj

**Review by ericLemanissier — CHANGES_REQUESTED** [review]

**Pull request closed without merging**